src/main/java/com/commercetools/pspadapter/notification/processor/impl/NotificationProcessorBase.java

```diff
@@ -54,17 +50,15 @@
                 );
     }
 
-    protected ImmutableList<UpdateAction<Payment>> createPaymentUpdates(@Nonnull Payment ctpPayment,
+    protected List<UpdateAction<Payment>> createPaymentUpdates(@Nonnull Payment ctpPayment,
                                                                         @Nonnull Event event) {
-        final ImmutableList.Builder<UpdateAction<Payment>> listBuilder = ImmutableList.builder();
-        listBuilder.add(createAddInterfaceInteractionAction(event));
-        Optional<ChangeTransactionState> changeTransactionOpt = createChangeTransactionState(ctpPayment);
-        if (changeTransactionOpt.isPresent()) {
-            listBuilder.add(changeTransactionOpt.get());
-        } else {
-            logger.warn("Notification event {} did not trigger change transaction state", event);
+        ArrayList<UpdateAction<Payment>> updateActions = new ArrayList<>();
+        updateActions.add(createAddInterfaceInteractionAction(event));
+        List<? extends UpdateAction<Payment>> updateAtions = createChangeTransactionState(ctpPayment);
+        if (!updateAtions.isEmpty()) {
```

Comment:
this `if` is redundant - JDK will do it for you implicitly.

Reply:
I'd even recommend to make it like this:
```java
ArrayList<UpdateAction<Payment>> updateActions = new ArrayList<>();
updateActions.add(createAddInterfaceInteractionAction(event));
updateActions.addAll(createChangeTransactionState(ctpPayment));
return updateActions;
```

of if you like Guava collection builders - in this case it may have sense to use it, for example, like this _inline_ solution:

```java
return ImmutableList.<UpdateAction<Payment>>builder()
        .add(createAddInterfaceInteractionAction(event))
        .addAll(createChangeTransactionState(ctpPayment))
        .build();
```

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/58/commits/24f840caf37122ca485127ab418ebd313d0090e8